sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/v2/state/StateDataSourceReadSuite.scala

```diff
@@ -796,4 +973,141 @@ abstract class StateDataSourceReadSuite extends StateDataSourceTestBase with Ass
       testForSide("right")
     }
   }
+
+  protected def testSnapshotNotFound(): Unit = {
+    withTempDir(tempDir => {
```

Comment:
nit: according to Databricks scala style, this should be `withTempDir { tempDir =>`, could save one indentation (curly brace)